fogros2/fogros2/kubernetes/gcp_kubernetes.py

```diff
@@ -0,0 +1,194 @@
+# FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+# INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+# DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE. REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE. THE SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY,
+# PROVIDED HEREUNDER IS PROVIDED "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE
+# MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+import json
+import os
+
+import subprocess
+import time
+import uuid
+import tempfile
+
+from ..util import extract_bash_column
+
+from ..cloud_instance import CloudInstance
+
+
+class GCPKubeInstance(CloudInstance):
+    """Kubernetes Implementation of CloudInstance."""
+
+    def __init__(
+            self,
+            container_image='ubuntu',
+            zone="us-central1-a",
+            mcpu=0,
+            mb=0,
+            **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.cloud_service_provider = "GKE"
+
+        id_ = str(uuid.uuid4())[0:8]
+        self._name = f'fog-{id_}-{self._name}'
+
+        self.zone = zone
+        self.type = f'{mcpu}mx{mb}Mb'
+        self.container_image = container_image
+
+        self._mcpu = mcpu
+        self._mmb = mb
+
+        self._working_dir = os.path.join(self._working_dir_base, self._name)
+        os.makedirs(self._working_dir, exist_ok=True)
+
+        # after config
+        self._ssh_key = None
+
+        self.create()
+
+    def create(self):
+        self.logger.info(f"Creating new Kubernetes Pod with name {self._name}")
+        self.create_compute_engine_instance()
+        self.info(flush_to_disk=True)
+        self.connect()
+        self.install_ros()
+        self.install_colcon()
+        self.install_cloud_dependencies()
+        self.push_ros_workspace()
+        self.info(flush_to_disk=True)
+        self._is_created = True
+
+    def info(self, flush_to_disk=True):
+        info_dict = super().info(flush_to_disk)
+        info_dict["compute_region"] = self.zone
+        info_dict["compute_instance_type"] = self.type
+        info_dict["compute_instance_id"] = self._name
+        if flush_to_disk:
+            with open(os.path.join(self._working_dir, "info"), "w+") as f:
+                json.dump(info_dict, f)
+        return info_dict
+
+    def create_service_pair(self, kube_wd: str, pub_key_path: str):
+        ssh_config: dict = json.loads(open(f'{kube_wd}/ssh.json').read())
+        vpn_config: dict = json.loads(open(f'{kube_wd}/vpn.json').read())
+        pod_config: dict = json.loads(open(f'{kube_wd}/pod.json').read())
+
+        # Configure the pod
+        pod_config['spec']['containers'][0]["image"] = self.container_image
+        pod_config['spec']['containers'][0]["name"] = self._name
```

Comment:
Perhaps change the quotations around "name" to be ' to match the rest of the strings.